bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/internal/Workbench.java

```diff
@@ -1483,6 +1490,82 @@ IWorkbenchWindow createWorkbenchWindow(IAdaptable input, IPerspectiveDescriptor
 		return result;
 	}
 
+	private void placeNearActiveShell(MWindow window) {
+		if (getDisplay() == null)
+			return;
+
+		Shell activeShell = getDisplay().getActiveShell();
+		if (activeShell == null)
+			return;
+
+		Rectangle shellBounds = activeShell.getBounds();
+		Monitor currentMonitor = findMonitorThatContains(shellBounds.x, shellBounds.y);
+
+		Rectangle currentMonitorBounds = currentMonitor.getBounds();
+
+		window.setX(calculateXInsideMonitor(shellBounds, currentMonitorBounds));
+
+		window.setY(calculateYInsideMonitor(shellBounds, currentMonitorBounds));
```

Comment:
The implemented functionality makes sense to me.
I only find the way in which it is implemented quite complicated to comprehend. For me it would make more sense to split up the flow as follows:
1. Calculate the desired value for the new window position (current shell position + 100 or so)
2. If necessary, adapt the new window position to fit into the monitor bounds

It could look something like this:
```java
final int offsetToExistingShell = 100;
Rectangle newShellBounds = activeShell.getBounds();
newShellBounds.x += offsetToExistingShell;
newShellBounds.y += offsetToExistingShell;
moveIntoBounds(newShellBounds, currentMonitorBounds);
window.setX(newShellBounds.x);
window.setY(newShellBounds.y);
```
with the helper method:
```java
private static void moveIntoBounds(Rectangle rectangleToMove, Rectangle bounds) {
	if (!bounds.union(rectangleToMove).equals(bounds)) {
		rectangleToMove.x = bounds.x + bounds.width - rectangleToMove.width;
		rectangleToMove.y = bounds.y + bounds.height - rectangleToMove.height;
	}
}
```

Reply:
Thank you for the feedback. I incorporated some of your changes with a few adaptations in the method `moveIntoBounds`. The changes I introduced make sure that:
* The new window only "bounces back" in the coordinate that left the bounds of the current monitor (_e.g._ if the new coordinates are too far down then only the _y_-coordinate will be corrected)
* The top-left corner of the new coordinates are always inside the bounds of the current monitor.

Hope this helps!